src/api/wiki.js

```diff
@@ -0,0 +1,333 @@
+                }
+                next(result);
+            });
+        } catch (e) {
+            next('');
+        }
+    };
+    if (ignore_cache_control) {
+        return await util.cacheFnIgnoreCacheControl(key, regenerate);
+    } else {
+        return await util.cacheFn(key, this.env.cache_control, regenerate);
+    }
+}
+
+/**
+ * Returns the page object for the specified page.
+ *
+ * @param {string} path
+ * @return {Promise<object>}
+ */
+async function getPage(path) {
+    var key = 'kuma:get_page:' + path.toLowerCase();
+    return JSON.parse(
+        await util.cacheFn(key, this.env.cache_control, next => {
+            var opts = {
+                method: 'GET',
+                headers: {
+                    'Cache-Control': this.env.cache_control
+                },
+                url: util.buildAbsoluteURL(path) + '$json'
+            };
+            try {
+                request(opts, function(err, resp, body) {
+                    let result;
+                    if (resp && 200 == resp.statusCode) {
+                        result = body;
+                    } else {
+                        result = '{}';
+                    }
+                    next(result);
+                });
+            } catch (e) {
+                next('{}');
+            }
+        })
+    );
+}
+
+/**
+ * Retrieve the full uri of a given wiki page.
+ *
+ * @param {string} path
+ * @param {string} query
+ * @return {string}
+ */
+function uri(path, query) {
+    const parts = url.parse(this.env.url);
+    var out = parts.protocol + '//' + parts.host + util.preparePath(path);
+    if (query) {
+        out += '?' + query;
+    }
+    return out;
+}
+
+/**
+ * Inserts a pages sub tree
+ * if reverse is non-zero, the sort is backward
+ * if ordered is true, the output is an <ol> instead of <ul>
+ *
+ * Special note: If ordered is true, pages whose locale differ from
+ * the current page's locale are omitted, to work around misplaced
+ * localizations showing up in navigation.
+ *
+ * @param {string} path
+ * @param {number} depth
+ * @param {number|boolean} self
+ * @param {number|boolean} reverse
+ * @param {number|boolean} ordered
+ */
+async function tree(path, depth, self, reverse, ordered) {
+    // If the path ends with a slash, remove it.
+    if (path.substr(-1, 1) === '/') {
+        path = path.slice(0, -1);
+    }
+
+    var pages = await this.page.subpages(path, depth, self);
+
+    if (reverse == 0) {
+        pages.sort(alphanumForward);
+    } else {
+        pages.sort(alphanumBackward);
+    }
+
+    return process_array(null, pages, ordered != 0, this.env.locale);
+
+    /**
+     * @param {string} t
+     * @return {string[]}
+     */
+    function chunkify(t) {
+        var tz = [],
+            x = 0,
+            y = -1,
+            n = 0,
+            i,
+            j;
+
+        while ((i = (j = t.charAt(x++)).charCodeAt(0))) {
+            var m = i == 46 || (i >= 48 && i <= 57);
+            if (m !== n) {
```

Comment:
error TS2367: This condition will always return 'true' since the types 'boolean' and 'number' have no overlap.

Reply:
I wonder if this code was a port of a C algorithm, and `n=0` above was intended to be `n=false`? Maybe the !== should have been a !=

But again, unless we can point to a specific content bug that is being caused by this, I don't want to go changing this legacy code for fear of causing unexpected changes to the output.